### DISCUSSION TITLE
How to handle unexpected exceptions? Is there any type of error event?

### DISCUSSION THREAD

**Discussion by Apocryphon-X** [Textualize/textual#3184] [Q&A] [answered]
Hi, I want to know if there is a way to handle _**unexpected**_ errors using the event system (the closest discussion I found is https://github.com/Textualize/textual/discussions/2819). I want to catch any non-critical error **without having exit the app**. This way the end user doesn't see the app crashing but instead it can see (i.e.) a nice notification/toast like this:
![image](https://github.com/Textualize/textual/assets/40130428/5c847751-1925-47da-ac14-9a4a8b54d72d)
> Note: I made the image above using `on_worker_state_changed` but I want to see if it's possible to handle errors that aren't from async workers this way.

**Comment by willmcgugan** [comment, marked as answer]
There isn't, and I wouldn't generally recommend it. If an error is unhandled, your app could be a an unpredictable state.

If you really want it, you could wrap your handlers in a try / except block. Or you could implement a decorator which catches exceptions and posts the notification.

**Reply by Apocryphon-X** [reply]
I see, thanks for the advice. Then I'm closing this as answered:)
